src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter((person) => person.sex === 'm');
+
+  men = century !== undefined
+    ? men.filter(man => Math.ceil(man.died / 100) === century)
+    : men;
+
+  const menAges = men.map(man => man.died - man.born);
+  const sumAges = menAges.reduce((a, b) => a + b, 0);
```

Comment:
U can shorten the map method. Alaso please move to separate function, because u use same in **calculateWomenAverageAge**
```suggestion
  const sumAges = menAges.reduce((a, b) => a + (b.died - b.born), 0);
```
![image](https://github.com/mate-academy/js_average-ages/assets/54936044/4719d946-f211-486e-a938-54ce5319857f)